playback-service/src/main/java/com/playback/PlaybackStore.java

```diff
@@ -0,0 +1,57 @@
+package com.playback;
+
+import com.playback.models.*;
+import com.playback.services.ContentSecurityPolicyService;
+import com.playback.services.CustomerInfoService;
+import com.playback.services.SteeringService;
+import com.playback.services.VideoMetadataService;
+import com.playback.standards.VideoResolution;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class PlaybackStore {
+    private final VideoMetadataService videoMetadataService;
+    private final CustomerInfoService customerInfoService;
+    private final ContentSecurityPolicyService contentSecurityPolicyService;
+    private final SteeringService steeringService;
+
+    public PlaybackStore(VideoMetadataService videoMetadataService, CustomerInfoService customerInfoService, ContentSecurityPolicyService contentSecurityPolicyService, SteeringService steeringService) {
+        this.videoMetadataService = videoMetadataService;
+        this.customerInfoService = customerInfoService;
+        this.contentSecurityPolicyService = contentSecurityPolicyService;
+        this.steeringService = steeringService;
+    }
+
+    public Optional<PlaybackManifest> request(String oauth, String title, Device device) {
+        if (!videoMetadataService.getMetadata().containsKey(title)) {
+            return Optional.empty();
+        }
+
+        final Optional<Collection<VideoMetadata>> videoMetadatas = customerInfoService
```

Comment:
Let's assume customerInfoService and videoMetadataService threw a checked exception, how would you modify this stream sequence?

Reply:
For `CustomerInfoService` we could make sure that we still return back an `Optional` in the case a checked exception that was thrown that would cause the customer to either be verified invalid or if we were unable to identify them. This try/catch can either exist in the `CustomerInfoService` class itself, or in a wrapping method.

As for the `VideoMetadataService` we could try to fit the contract of putting the `VideoMetadata` into an `Optional`. We could achieve this with a try/catch wrapping method or within the `VideoMetadataService` class itself. That way we can map over it if it was successfully fetched otherwise we'd end up terminating with an empty `Collection<VideoMetadata>`

Reply:
We'd need then to also handle that an empty `Collection<VideoMetadata>` most likely results in a `PlaybackManifest` error or the device being unable to play the requested title.